src/client/tst/IntermittentProducerAutomaticStreamingTest.cpp

```diff
@@ -0,0 +1,192 @@
+#include "ClientTestFixture.h"
+
+using ::testing::WithParamInterface;
+using ::testing::Bool;
+using ::testing::Values;
+using ::testing::Combine;
+
+class IntermittentProducerAutomaticStreamingTest : public ClientTestBase,
+                                    public WithParamInterface< ::std::tuple<UINT64, UINT32> > {
+protected:
+
+    void SetUp() {
+        ClientTestBase::SetUp(false);
+        UINT64 callbackInterval;
+        UINT32 clientInfoVersion;
+        std::tie(callbackInterval, clientInfoVersion) = GetParam();
+        mDeviceInfo.clientInfo.version = CLIENT_INFO_CURRENT_VERSION;
+        mDeviceInfo.clientInfo.callbackPeriod = (UINT64) callbackInterval * HUNDREDS_OF_NANOS_IN_A_MILLISECOND;
+        mDeviceInfo.clientInfo.automaticStreamingFlags = AUTOMATIC_STREAMING_INTERMITTENT_PRODUCER;
+    }
+
+};
+
+TEST_P(IntermittentProducerAutomaticStreamingTest, ValidateTimerInvokedBeforeTime) {
+    // Create new client so param value of callbackPeriod can be applied
+    ASSERT_EQ(STATUS_SUCCESS, CreateClient());
+
+    PKinesisVideoClient client = FROM_CLIENT_HANDLE(mClientHandle);
+
+    EXPECT_EQ((int)client->timerCallbackInvocationCount, 0);
+
+    // Create synchronously
+    CreateStreamSync();
+
+    // Produce a frame
+    BYTE temp[100];
+    Frame frame;
+    frame.trackId = 1;
+    frame.size = SIZEOF(temp);
+    frame.duration = TEST_FRAME_DURATION;
+    frame.index = 0;
+    frame.flags = FRAME_FLAG_KEY_FRAME;
+    frame.presentationTs = 0;
+    frame.decodingTs = 0;
+    frame.frameData = temp;
+    EXPECT_EQ(STATUS_SUCCESS, putKinesisVideoFrame(mStreamHandle, &frame));
+
+    THREAD_SLEEP(500 * HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+
+    EXPECT_EQ((int)client->timerCallbackInvocationCount, 1);
+}
+
+TEST_P(IntermittentProducerAutomaticStreamingTest, ValidateTimerInvokedAfterFirstPeriod) {
+    // Create new client so param value of callbackPeriod can be applied
+    ASSERT_EQ(STATUS_SUCCESS, CreateClient());
+
+    PKinesisVideoClient client = FROM_CLIENT_HANDLE(mClientHandle);
+
+    // Create synchronously
+    CreateStreamSync();
+
+    // Produce a frame
+    BYTE temp[100];
+    Frame frame;
+    frame.trackId = 1;
+    frame.size = SIZEOF(temp);
+    frame.duration = TEST_FRAME_DURATION;
+    frame.index = 0;
+    frame.flags = FRAME_FLAG_KEY_FRAME;
+    frame.presentationTs = 0;
+    frame.decodingTs = 0;
+    frame.frameData = temp;
+    EXPECT_EQ(STATUS_SUCCESS, putKinesisVideoFrame(mStreamHandle, &frame));
+
+    THREAD_SLEEP(200 * HUNDREDS_OF_NANOS_IN_A_MILLISECOND + 1.25 * mDeviceInfo.clientInfo.callbackPeriod);
+
+    EXPECT_EQ((int)client->timerCallbackInvocationCount, 2);
+}
+
+/*
+This test fails, if I try to do only stopKinesisVideoStreamSync it fails there because the close
```

Comment:
We can model the true stopStreamSync in producer c SDK tests. But for here, it's harder to do. What you could do is submit the last persisted ACK or similar to trigger the stopSync to return